includes/Core/Storage/Has_Connected_Admins.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Storage\Has_Connected_Admins
+ *
+ * @package   Google\Site_Kit\Core\Storage
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Storage;
+
+use Google\Site_Kit\Core\Authentication\Clients\OAuth_Client;
+use Google\Site_Kit\Core\Storage\Options_Interface;
+use Google\Site_Kit\Core\Storage\Setting;
+use Google\Site_Kit\Core\Storage\User_Options_Interface;
+
+/**
+ * Has_Connected_Admins class.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class Has_Connected_Admins extends Setting {
+
+	/**
+	 * The option_name for this setting.
+	 */
+	const OPTION = 'googlesitekit_has_connected_admins';
+
+	/**
+	 * User options instance implementing User_Options_Interface.
+	 *
+	 * @since n.e.x.t
+	 * @var User_Options_Interface
+	 */
+	protected $user_options;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Options_Interface      $options      Options instance.
+	 * @param User_Options_Interface $user_options User options instance.
+	 */
+	public function __construct( Options_Interface $options, User_Options_Interface $user_options ) {
+		parent::__construct( $options );
+		$this->user_options = $user_options;
+	}
+
+	/**
+	 * Registers the setting in WordPress.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function register() {
+		parent::register();
+
+		$access_token_meta_key = $this->user_options->get_meta_key( OAuth_Client::OPTION_ACCESS_TOKEN );
+
```

Comment:
There is a bit of a bug here in that if the option is already set as `false` and then a user is connected, it would remain `false` until a user first disconnected or the option was otherwise deleted causing the query to run again. Let's fix that.

All we should need to do here is add another action on `added_user_meta` similar to the action we have below on the delete action. If the meta key is for the access token key, and `user_can( $object_id, 'administrator' )` then we can call `$this->set( true )`.

Reply:
Updated.

Reply:
It looks like maybe I wasn't clear about what needed to change here; my apologies if that is the case!

However, resetting the setting any time the meta is changed at all is not quite what we want.

Consider this for each:
* `added_user_meta` - if the user the meta is being added to is an administrator, we already know the option can be set to `true`, so there is no need to delete it. If the option is already set to `true` it will be a no-op as WP will only update the option if the new value is different.
* `updated_user_meta` - we don't need to take any action in this case since we're only concerned about the meta existing (which for this action it always will), just like in the user query, so this would cause the option value to be deleted and re-queried fairly often since access tokens are only valid for one hour before they're refreshed.
* `deleted_user_meta` - we don't know if there are any other admins with tokens at this time, and we wouldn't without querying again or maybe changing the value stored to be a list of IDs, but that isn't necessary. Since we also need to populate the option if it isn't set yet on-demand we can delete it here to make sure it is refreshed on the next time it is requested.

If I'm missing something here please let me know, otherwise we only need the two actions here with their own specific behavior rather than resetting when there is any change 🙂

Reply:
Updated.